### PR TITLE
init QRcode support.

### PR DESCRIPTION
**Pull request by Lunatixz**
Hi, I started initial QR support... branch is working but I'm having two small issues.

1. for some reason I can't get a accurate width or height using getWidth, getHeight... any thoughts?
2. is there a easy way to get center screen x.y position? 

The plugin now requires the below module for QR, I plan on submitting all my repo plugins/modules to the official repo.

https://github.com/Lunatixz/XBMC_Addons/tree/master/script.module.pyqrcode

### CODE REVIEW

**Comment by Lunatixz** [comment]
Any suggestions on aesthetics? I'd like to avoid requiring a custom xml. I'd like to use what's available to Kodi's current skin.

**Comment by ronie** [comment]
> for some reason I can't get a accurate width or height using getWidth, getHeight... any thoughts?

nope, not really. 
i don't know how it even is supposed to work...
skins don't define a width/height for their dialogs.

> is there a easy way to get center screen x.y position?

perhaps you can use the System.ScreenWidth / System.ScreenHeight infolabels?

**Comment by ronie** [comment]
> Any suggestions on aesthetics? I'd like to avoid requiring a custom xml

would be difficult.. we don't really have a suitable dialog in kodi for text + image.

**Comment by Lunatixz** [comment]
Is the current gui method okay? I'm calling a fixed dialog and filling it's textbox... then overlay a control image.

File: default.py [hunk 1]
```diff
@@ -22,13 +24,47 @@
...
 
+class QRCode(xbmcgui.WindowDialog):
+    def __init__(self, url, message):        
+        url = pyqrcode.create(url)
+        url.png(IMAGEFILE, scale=10, module_color=(255, 255, 255, 255), background=(0, 0, 0, 255)) 
+        self.showWindow(message)
+ 
+    def showWindow(self, text, heading=ADDONNAME):
+        id = 10147
+        xbmc.executebuiltin('ActivateWindow(%d)' % id)
```
Comment (Lunatixz) [inline default.py]: I tried, I had issues overlay the controlimage (ie QR code) while dealing with another open domodel (ie textview).

**Comment by Lunatixz** [comment]
I'd like to find a clean solution for displaying the QR code.... any suggestions ?

**Comment by ronie** [comment]
hmmm... i think the only solid solution would by to create a script xml dialog.
i always prefer the WindowXMLDialog over trying to create a dialog in pure python.

the xml way is more flexible, has more options and allows skinners to create a custom xml, if they want to.

**Comment by Lunatixz** [comment]
Okay I'll go with a custom xml... do you know of a way I can utilize the current skins dialog textures in order to preserve the users look and feel? I don't believe there is a standard naming scheme for textures?

Thanks

**Comment by ronie** [comment]
nope you can't and you shouldn't ;-)
you need to include all needed graphics in the addon.

best practice would be to prefix the filenames of the imagesd, so they don't clash with anything that might be included in a skin.
for instance: https://github.com/ronie/script.globalsearch/tree/master/resources/skins/default/media

**Comment by Lunatixz** [comment]
Yes... I understand a standalone package has to be available to the script : )

I guess there is no way to seamlessly incorporate a script into the look and feel of kodi without creating individual skin xmls... 

I'll create a custom xml to look like estuary, and leave it up to skinners to support?

Thanks for the feedback...

**Comment by ronie** [comment]
yup, that would be it.

**Comment by Lunatixz** [comment]
Can't see your comment about uuid... Only reason its required? was to avoid using a cached qr image.
If you prefer I can name the image the same as the url.

**Comment by ronie** [comment]
thx!

will test it and might make a few cosmetic changes.

**Comment by Lunatixz** [comment]
Okay, I'll take note of your cosmetic changes... I'm not very skilled when it comes to skinning : )

**Comment by ronie** [comment]
yeah, want to get rid of setting/clearing window properties.

**Comment by Lunatixz** [comment]
You prefer setting direct control ids? Is that the correct method?

As for clearing windows, I like to tidying up after a plugin execution. Guess this wouldn't be needed if I used control ids ; )

If this is the case... def showresults can pass the url, message argument to the QRcode class...

**Pull request merged**